### PR TITLE
log4js integration

### PR DESCRIPTION
**Pull request by hiddentao**
This replaces `console.log` calls with a [log4js](https://github.com/nomiddlename/log4js-node) integration and adds two command-line flags (`--loglevel` and `--logfile`) to be able to customize logging verbosity at runtime.

screen shot 2016-04-18 at 12 26 45

Benefits:
* Console output is now timestamped and categorized - with the ability to customize the format in a single place (the `logger.js` file). We could also add a CLI arg to override the pattern if deemed necessary.
* We can now add debug and trace logging in many more parts of the app without automatically clogging up the console output, yet retaining the ability to show these logs by setting the right CLI argument.
* We can set the logging threshold using `--loglevel <level>`. Possible values are `trace`, `debug`, `info`, `warn`, `error`. Default is `info`.
* We can have log4js log to a file using the `--logfile <path to file>` arg.
* Log4js output formatter automatically handles things like `Error` objects well.

I haven't added this logging to the Meteor "interface" app yet but there's no reason we can't do that.

### CODE REVIEW

**Comment by hiddentao** [comment]
Sorry, wrong branch.

File: main.js [hunk 1]
```diff
@@ -100,6 +108,8 @@ if(global.mode === 'wallet') {
 
 // MIST
 } else {
+    log.info('Starting in Mist mode');
+
     global.interfaceAppUrl = global.interfacePopupsUrl = (global.production)
         ? 'file://' + __dirname + '/interface/index.html'
         : 'http://localhost:3000';
```
Comment (aliyathoummavongsa) [inline main.js]: readme-edits

File: main.js [hunk 2]
```diff
@@ -141,7 +154,7 @@ app.on('window-all-closed', function() {
 
 // Listen to custom protocole incoming messages, needs registering of URL schemes
 app.on('open-url', function (e, url) {
-    console.log('Open URL', url);
```
Comment (aliyathoummavongsa) [inline main.js]: readme-edits
\

File: main.js [hunk 3]
```diff
@@ -157,7 +170,7 @@ app.on('before-quit', function(event){
     // CLEAR open IPC sockets to geth
     _.each(global.sockets, function(socket){
         if(socket) {
-            console.log('Closing Socket ', socket.id);
```
Comment (aliyathoummavongsa) [inline main.js]: readme-edits

Comment (aliyathoummavongsa) [inline main.js]: readme-edits

File: main.js [hunk 4]
```diff
@@ -194,7 +207,7 @@ var appStartWindow;
 var nodeType = 'geth';
 var logFunction = function(data) {
     data = data.toString().replace(/[\r\n]+/,'');
-    console.log('NODE LOG:', data);
```
Comment (aliyathoummavongsa) [inline main.js]: readme-edits

**Pull request closed without merging**